From 9770cad990864af88c1f55ec31302ccd0e849a75 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 12:01:30 -0400
Subject: [ethernet] mlx4: Run loopback test only when port is up

Message-id: <0264aa0ffcd34d091d925e667ced5396384fd78c.1410425016.git.avadai@redhat.com>
Patchwork-id: 93737
O-Subject: [Patch RHEL7.1 V1 154/197] net/mlx4_en: Run loopback test only when port is up
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Eugenia Emantayev <eugenia@mellanox.com>

Loopback can't work when port is down.

Signed-off-by: Eugenia Emantayev <eugenia@mellanox.com>

Signed-off-by: Amir Vadai <amirv@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 4359db1e0d9f763dfcffbf23fb4a19341aad9b6b)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/mellanox/mlx4/en_selftest.c b/drivers/net/ethernet/mellanox/mlx4/en_selftest.c
index 03e5f6a..49d5afc 100644
--- a/drivers/net/ethernet/mellanox/mlx4/en_selftest.c
+++ b/drivers/net/ethernet/mellanox/mlx4/en_selftest.c
@@ -159,7 +159,8 @@ void mlx4_en_ex_selftest(struct net_device *dev, u32 *flags, u64 *buf)
   if (priv->mdev->dev->caps.flags &
      MLX4_DEV_CAP_FLAG_UC_LOOPBACK) {
    buf[3] = mlx4_en_test_registers(priv);
-   buf[4] = mlx4_en_test_loopback(priv);
+   if (priv->port_up)
+    buf[4] = mlx4_en_test_loopback(priv);
   }
 
   if (carrier_ok)
-- 
1.7.1